platform/comm/src/main/java/com/fmicodes/comm/services/GoogleMapsService.java

```diff
@@ -0,0 +1,109 @@
+
+        try (Response response = client.newCall(request).execute()) {
+            String jsonResponse = Objects.requireNonNull(response.body()).string();
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(jsonResponse);
+
+            List<RestaurantInfo> restaurants = new ArrayList<>();
+
+            if (jsonNode.has("results")) {
+                for (JsonNode result : jsonNode.get("results")) {
+
+                    if (restaurants.size() > 3) {
+                        break;
+                    }
+
+                    RestaurantInfo restaurant = new RestaurantInfo();
+
+                    if (result.has("name")) {
+                        restaurant.setName(result.get("name").asText());
+                    }
+
+                    if (result.has("vicinity")) {
+                        restaurant.setLocation(result.get("vicinity").asText());
+                    }
+
+                    if (result.has("rating")) {
+                        restaurant.setRating(result.get("rating").asDouble());
+                    }
+
+                    if (result.has("price_level")) {
+                        restaurant.setPriceLevel(result.get("price_level").asInt());
+                    }
+
+                    if (result.has("user_ratings_total")) {
+                        restaurant.setUserRatingsTotal(result.get("user_ratings_total").asInt());
+                    }
+
+                    if (result.has("opening_hours") && result.get("opening_hours").has("open_now")) {
+                        boolean isOpen = result.get("opening_hours").get("open_now").asBoolean();
+                        restaurant.setOpeningHours(isOpen ? "Open now" : "Closed");
+                    }
+
+                    if (result.has("place_id")) {
+                        String placeId = result.get("place_id").asText();
+                        restaurant.setGoogleMapsLink("https://www.google.com/maps/place/?q=place_id:" + placeId);
+                    }
+
+
+                    if (result.has("photos")) {
+                        JsonNode photo = result.get("photos").get(0);
+                        int photoWidth = 1000;
+                        int photoHeight = 1000;
+                        if (photo.has("width")) {
+                            photoWidth = photo.get("width").asInt();
+                        }
+                        if (photo.has("height")) {
+                            photoHeight = photo.get("height").asInt();
+                        }
+
+                        if (photo.has("photo_reference")) {
+                            String photoReference = photo.get("photo_reference").asText();
+                            restaurant.setPhotoUrl(getPhotoUrl(photoReference, photoHeight, photoWidth));
+                        }
+                    }
+
+                    restaurants.add(restaurant);
+                }
+            }
+
+            return restaurants;
```

Comment:
Move return out of try block and at the end of the method

Reply:
taken into account